newIDE/app/src/Profile/Subscription/SubscriptionDialog.js

```diff
@@ -121,6 +131,25 @@ const cancelAndChangeWithValidRedeemedCodeConfirmationTexts = {
   maxWidth: 'sm',
 };
 
+const getSubscriptionPricingSystemPeriod = (
+  subscription: ?Subscription,
+  subscriptionPlansWithPricingSystems: ?(SubscriptionPlanWithPricingSystems[])
+): null | 'year' | 'month' => {
+  if (!subscription || !subscriptionPlansWithPricingSystems) return null;
+  for (let subscriptionPlanWithPricingSystems of subscriptionPlansWithPricingSystems) {
+    for (let pricingSystem of subscriptionPlanWithPricingSystems.pricingSystems) {
+      if (pricingSystem.id === subscription.pricingSystemId) {
+        if (pricingSystem.period === 'week') {
```

Comment:
those for/ifs are a bit hard to read, or maybe it's just me.
Wondering if it would be easier to read with a xxx.filter() or xxx.find() instead

Reply:
I'll see if I can simplify it